transformer_engine/pytorch/tensor/float8_tensor.py

```diff
@@ -166,6 +167,167 @@ def create_tensor_from_data(
         )
 
 
+class Float8CurrentScalingQuantizer(Quantizer):
+    """Builder class for FP8 tensors with per-tensor current scaling
+
+    High-precision tensors (e.g. in FP32 or BF16) are quantized by
+    multiplying with a scaling factor and casting to FP8. The max-abs
+    value ("amax") in the tensor is computed directly by scanning the input
+    high-precision tensor, without the need of any history window.
+
+    Unlike delayed scaling, scale and amax tensors are not needed to initialize the
+    quantizer, becuse they are simply GPU buffers that will be filled by current
+    scaling quantization kernels, instead of using values taken from delayed scaling
+    history window. Therefore, device parameter is needed for tensor allocation.
+
+    Both Float8CurrentScalingQuantizer and Float8Quantizer produces Float8Tensor,
+    because they are both per-tensor scaling, ie. one scaling factor per tensor.
```

Comment:
`Float8CurrentScalingQuantizer` reusing significan portion of `Float8Quantizer`, what about introducing a common base class, e.g. `PerTensorQuantizer(Quantizer)`? This can help reducing code duplication (there are several common attributes and methods).